api/ranked1v1.py

```diff
@@ -46,20 +46,82 @@ def ranked1v1():
         where += " l.login LIKE %(player)s"
         args['player'] = '%' + player + '%'
 
-    return fetch_data(Ranked1v1Schema(), TABLE, SELECT_EXPRESSIONS, MAX_PAGE_SIZE, request, sort='-rating',
+    select = append_select_expression()
+
+    return fetch_data(Ranked1v1Schema(), TABLE1V1, select, MAX_PAGE_SIZE, request, sort='-rating',
                       args=args, where=where)
 
+@app.route('/rating/<string:rating_type>')
+def rating_type(rating_type):
+    if request.values.get('sort'):
+        raise InvalidUsage('Sorting is not supported for ranked1v1')
+
+    page = int(request.values.get('page[number]', 1))
+    page_size = int(request.values.get('page[size]', MAX_PAGE_SIZE))
+    player = request.args.get('filter[player]')
+    row_num = (page - 1) * page_size
+    select = SELECT_EXPRESSIONS
+
+    args = {'row_num': row_num}
+
+    where = ''
+    active_filter = request.values.get('filter[is_active]')
+    if active_filter:
+        where += 'is_active = ' + ('1' if active_filter.lower() == 'true' else '0') + ' AND r.numGames > 0'
+
+    if player:
+        where += " l.login LIKE %(player)s"
+        args['player'] = '%' + player + '%'
+
+    if rating_type == '1v1':
+        table = TABLE1V1
+        select = append_select_expression()
+    elif rating_type == 'global':
+        table = TABLEGLOBAL
+    else:
+        raise InvalidUsage('The rating type should be either 1v1 or global.')
```

Comment:
Instead of having logic here, I would create two static routes which use a common method